org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinLateinitFilter.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Fabian Mastenbroek - implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+/**
+ * Filters Kotlin lateinit properties that generate bytecode of which parts will always be missed
+ * under normal circumstances.
```

Comment:
> Filters Kotlin lateinit properties that generate bytecode

IMO bytecode is generated by Kotlin compiler, not by properties, and properties per se are not filtered.

> will always be missed under normal circumstances

This is target of our filters, so seems redundant.

What about rewording into something like

```
/**
 * Filters branch in bytecode that Kotlin compiler generates for reading from
 * <code>lateinit</code> properties.
 */
```

? I'm thinking to add similar sentence into changelog.

Reply:
That reads better indeed